src/net/sourceforge/kolmafia/KoLAdventure.java

```diff
@@ -2132,7 +2132,8 @@ public boolean canAdventure() {
 
     if (this.zone.startsWith("PirateRealm")) {
       // *** You have limited turns available per day.
-      return Preferences.getBoolean("prAlways") || Preferences.getBoolean("_prToday");
+      return (Preferences.getBoolean("prAlways") || Preferences.getBoolean("_prToday"))
+          && Preferences.getString("_LastPirateRealmIsland").contains(this.adventureName);
```

Comment:
is there a case where `equals` wouldn't work?

Reply:
Not that I'm aware of, but I'm yet to check logs nontrivially

Reply:
update: @DamianDominoDavis just scraped some logs and `equals` is good